src/map/map.cpp

```diff
@@ -2685,13 +2685,11 @@ int map_addinstancemap(const char *name, unsigned short instance_id)
 	struct map_data *src_map = map_getmapdata(src_m);
 	struct map_data *dst_map = map_getmapdata(dst_m);
 
-	strcpy(iname, name);
-
 	// Alter the name
 	// Due to this being custom we only worry about preserving as many characters as necessary for accurate map distinguishing
 	// This also allows us to maintain complete independence with main map functions
 	if((strchr(iname,'@') == NULL) && strlen(iname) > 8) {
-		memmove(iname, iname+(strlen(iname)-9), strlen(iname));
+		memmove((void*)iname, iname+(strlen(iname)-9), strlen(iname));
```

Comment:
Use C++ Type Casting (`const_cast<char*>(iname)`), In my personal opinion, I think you should just create a temp pointer to the first character that you want to read,  like `const char* subSTR= name + offset`